pages/overview/what-is-jsonschema.md

```diff
@@ -6,36 +6,285 @@ title: What is JSON Schema?
+}
+
+try:
+    validate(data, schema)
+    print("Valid JSON object.")
+except Exception as e:
+    print("Invalid JSON object:", e)
+
+```
+
+In the above code snippet, we defined a JSON Schema that describes the expected structure and constraints of the JSON data. We use the `jsonschema` library to validate the `data` against the `schema`. If the data doesn't conform to the schema, an exception is raised, providing detailed information about the failure.
+
+By using JSON Schema, we can easily define and enforce constraints, making the validation process more robust and manageable. It improves the readability of the code and reduces the chances of data-related issues.
+
+
+### Steps to Validate JSON Data Using JSON Schema
+
+1. **Create the schema**
+
+    You begin by defining the rules in a JSON Schema document using [keywords](https://json-schema.org/draft/2020-12/json-schema-validation.html#name-validation-keywords-for-any).
+
+    * You can use `type` to specify what types of data are valid for a given key. You can specify either a single type, like `string`, or an array of possible types, like `string`, `number`, or `integer`.
+    * You can also enforce formats using regular expressions with the `pattern` keyword.
+    * There are countless ways to describe specific data types and formats that your data must follow.
+
+    Now that the rules have been created, you can use these rules to create a JSON Schema that represents the expected structure and format of your JSON data.
+
+    For example, let the following file be `schema.json`:
+
+    ```
+    schema = {
+        "type": "object",
+        "properties": {
+            "name": {
+            "type": ["string", "null"],
+            "pattern": "^[A-Za-z]+$"
+            },
+            "age": {
+            "type": "integer"
+            },
+            "email": {
+            "type": "string"
+            }
+        },
+        "required": ["name", "age", "email"]
+    }
+    ```
+
+    * In the above schema, `name` can either be a `string` or `null`, `age` can only be an `integer`, `email` can only be a `string`, and the entire schema is an `object`.
+    * Here, `name` can only contain alphabetical characters. 
+    * You can make some properties mandatory to enter by using the `required` keyword. Here, `name`, `age`, and `email`, all are required to create an instance of the schema.
+
+2. **Choose a validator**
+
+    Select a JSON Schema [validator](https://json-schema.org/implementations.html#validators) that is supported in your chosen programming language or context. There are JSON Schema validation tools for nearly every modern programming language, as well as language-agnostic command-line and browser tools.
+
+3. **Validate the instance**
+    
+    Use the selected validator to validate your instance against the created JSON Schema.
+
+    Here's an example using the [jsonschema](https://github.com/python-jsonschema/jsonschema) Python library:
```

Comment:
Can we copy this to other examples of python code? =]

Reply:
I'm not sure I understand. Could you please clarify where exactly you would like to copy the python code? The 'Miscellaneous Examples' section? 😅

Reply:
There are multiple examples. This is the only one (I think) that tells you it is in Python and which library it is using. Can all the examples add this line please?

If someone reads the page from start to finish, it would not be needed. However, you can be sure that many will not, and will start at different sections they have been linked to.

Reply:
Okay, yes. I'll add this to other examples as well to make it more clear. Thanks for pointing this out!